src/RemoteAPI/related/class-related-proxy.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * Parsely Related REST API Proxy
+ *
+ * @package Parsely
+ * @since 3.2.0
+ */
+
+declare(strict_types=1);
+
+namespace Parsely\RemoteAPI;
+
+use Parsely\Parsely;
+use WP_Error;
+
+/**
+ * Proxy for the /related endpoint.
+ *
+ * @since 3.2.0
+ */
+class Related_Proxy implements Proxy {
+	const RELATED_API_ENDPOINT = 'https://api.parsely.com/v2/related';
+
+	/**
+	 * Parsely Instance.
+	 *
+	 * @var Parsely
+	 */
+	private $parsely;
+
+	/**
+	 * The query arguments to send to the remote API.
+	 *
+	 * @var array
+	 */
+	private $query;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param Parsely $parsely Parsely instance.
+	 * @param array   $query The query arguments to send to the remote API.
+	 * @since 3.2.0
+	 */
+	public function __construct( Parsely $parsely, array $query ) {
+		$this->parsely = $parsely;
+		$this->query   = $query;
+	}
+
+	/**
+	 * Accessor for private query var.
+	 *
+	 * @since 3.2.0
+	 */
+	public function get_query() {
```

Comment:
I can't think of a better way to generate a unique cache key, so I think you're right that this needs to be exposed. As such, should it be added to the interface as well?

Reply:
I changed the approach, so this comment is no longer relevant.